dlg/dropmake/web/lg_web.py

```diff
@@ -343,6 +351,81 @@ def gen_pgt():
         trace_msg = traceback.format_exc()
         return "Graph partition exception {1}: {0}".format(trace_msg, lg_name)
 
+@route('/gen_pgt', method='POST')
+def gen_pgt_post():
+    """
+    Translating Logical Graphs to Physical Graphs.
+    Differs from get_pgt above by the fact that the logical graph data is POSTed
+    to this route in a HTTP form, whereas gen_pgt loads the logical graph data
+    from a local file
+    """
+    # Retrieve the graph name.
+    lg_name = request.forms.get('lg_name')
+
+    # Retrieve json data.
+    json_string = request.forms.get('json_data')
+    logical_graph = json.loads(json.loads(json_string))
+
+    try:
+        # Save graph
+        new_path = save(lg_name, logical_graph)
+
+        # Unrolling LG to PGT.
+        pgt = unroll(new_path)
+
+        # Define partitioning parameters.
+        algo = request.forms.get('algo', 'none')
+        num_partitions = request.forms.get('num_par', default=1, type=int)
+        num_islands = request.forms.get('num_islands', default=0, type=int)
+        par_label = request.forms.get('par_label', 'Partition')
+
+        # Build a map with extra parameters, more specific to some algorithms.
+        algo_params = {}
+        for name, typ in ALGO_PARAMS:
+            if name in request.forms:
+                algo_params[name] = request.forms.get(name, type=typ)
+
+        # Partition the PGT
+        pgt = partition(pgt, algo=algo, num_partitions=num_partitions,
+                        num_islands=num_islands, partition_label=par_label,
+                        show_gojs=True,
+                        **algo_params)
+
+        pgt_id = pg_mgr.add_pgt(pgt, lg_name)
+
+        part_info = ' - '.join(['{0}:{1}'.format(k, v) for k, v in pgt.result().items()])
```

Comment:
I think from this point onward you want to do this only for the GET version of the request, rigth? I would imagine in the POST case you just want a 200 return code to signal success, in which case you'll just want to return from this method without further ado.